src/rhp.rs

```diff
@@ -12,21 +13,63 @@ pub fn sector_root(sector: &[u8]) -> Hash256 {
     let mut params = Params::new();
     params.hash_length(32);
 
-    let mut tree_hashes = sector
-        .par_chunks_exact(SEGMENT_SIZE)
-        .map(|chunk| sum_leaf(&params, chunk))
-        .collect::<Vec<_>>();
+    let mut tree_hashes = vec![[0; 32]; SECTOR_SIZE / SEGMENT_SIZE];
```

Comment:
that's a big vec -- 2 MiB! Have you tried something like:
```rust
let mut tree_hashes = vec![[0; 32]; 64];
tree_hashes
    .par_chunks_exact_mut(1)
    .enumerate()
    .for_each(|(i, chunk)| {
        // ... hash a group of SECTOR_SIZE/64 segments into their Merkle root ...
    })

// same as before
let mut chunk_size = 4;
while chunk_size <= tree_hashes.len() {
    ...
```

Really though the Right Thing To Do is for me to learn Rust and optimize the `Accumulator` to use `blake2b_simd` 😩